### PR TITLE
Separate "development" and "production" configurations

### PR DESCRIPTION
**Pull request by thaJeztah**
This patch updates the default configuration to match a "development"
situation, and introduces build-options to produce a "production"
build.

By default (dev environment):

- Google Analytics / GTM and PollDaddy are disabled
- SASS builds non-minified stylesheets (for easier readabililty)
- Excludes "enterprise" stubs

Building a "production" build locally is still possible by overriding
the `JEKYLL_ENV` build-arg;

    JEKYLL_ENV=production docker-compose  up --build

### CODE REVIEW

File: .github/workflows/build-master.yml [hunk 1]
```diff
@@ -17,7 +17,7 @@ jobs:
         run: docker version && docker info
       - uses: actions/checkout@v2
       - name: build current docs
-        run: docker build --target=deploy-source --output=./_site .
+        run: docker build --build-arg JEKYLL_ENV=production --target=deploy-source --output=./_site .
```
Comment (thaJeztah) [inline .github/workflows/build-master.yml:20]: Wasn't sure what environment we want "master" to be built for; perhaps this should also be "development"?

**Comment by usha-mandya** [comment]
Thank you @thaJeztah. Does it change the way we publish docs?

**Comment by thaJeztah** [comment]
@usha-mandya no, should not change things; 

- when you build the docs locally, it builds them as "development"
- same for "PR" builds
- same for "Netlify"

On "master" builds and on "build-published" builds, it will build for "production" (so include Google analytics, and compressed CSS)

Not sure if "master" builds should also build "production"; perhaps not, as that also pushes images to docker hub for users to run it themselves, and to our staging domain @StefanScherer wdyt?

**Comment by thaJeztah** [comment]
I may need to do one more tweak that I didn't get to work easily (but have some ideas), and that's to disable indexing for development builds

File: .github/workflows/build-master.yml (hunk 1 above)
Comment (usha-mandya) [reply .github/workflows/build-master.yml:20]: I believe it makes sense to keep master builds as dev builds.

**Comment by usha-mandya** [comment]
> I may need to do one more tweak that I didn't get to work easily (but have some ideas), and that's to disable indexing for development builds

Sure. Are you planning to include it as part of this PR, or as a follow up PR?

**Comment by thaJeztah** [comment]
> Sure. Are you planning to include it as part of this PR, or as a follow up PR?

Let me do so as a follow-up; disable indexing also removes pages from the search auto-complete, and I guess it's useful to still have that working as usual 🤔

File: .github/workflows/build-master.yml (hunk 1 above)
Comment (StefanScherer) [reply .github/workflows/build-master.yml:20]: Yes, master should be still "development" without polldaddy, ...

Comment (thaJeztah) [reply .github/workflows/build-master.yml:20]: oh, let me update that 👍

**Review by usha-mandya — DISMISSED** [review]
LGTM. Thank you @thaJeztah . You are a star!!

**Comment by thaJeztah** [comment]
@usha-mandya I updated and removed the "production" build from master builds, so only the "build-published" will now produce "production" builds 👍

**Review by StefanScherer — APPROVED** [review]
LGTM

**Review by usha-mandya — APPROVED** [review]

**Pull request merged**